hail/python/hail/ir/ir.py

```diff
@@ -548,6 +548,42 @@ def _compute_type(self, env, agg_env):
         self._type = tarray(tint32)
 
 
+class MakeNDArray(IR):
+    @typecheck_method(flags=IR, shape=IR, offset=IR, strides=IR, data=IR)
```

Comment:
Hmm.  I don't think the constructor should have stride or offset, and flag should just be the ordering (e.g. row_order: bool).  This feels like you're confusing the interface and the implementation (e.g. offset and strides is to support slicing).  Personally, I'd write it `data, shape, row_order`.